### PR TITLE
Add auto-compound and hdro tvl

### PR DESCRIPTION
**Pull request by hydro-developer**
**NOTE**

#### Please enable "Allow edits by maintainers" while putting up the PR.

---

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
3. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
4. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
5. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
6. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---

### CODE REVIEW

**Comment by llamatester** [comment]
Error while running adapter at projects/hydro-protocol: 
        
 
  
    
    
    Error: Protocol doesn't have total tvl, make sure to export a tvl key either on the main object or in one of the chains
    Truncated error stack:
    
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:192:11

**Comment by llamatester** [comment]
Error while running adapter at : 
        
 
  > Please revert changes to package.json / package-lock.json

**Comment by llamatester** [comment]
Error while running adapter at projects/hydro-protocol: 
        
 
  
    
    
    Error: Protocol doesn't have total tvl, make sure to export a tvl key either on the main object or in one of the chains
    Truncated error stack:
    
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:192:11

File: projects/hydro-protocol/index.js [hunk 1]
```diff
@@ -1,10 +1,65 @@
...
+
+const injectiveSpotApi = new IndexerGrpcSpotApi("https://sentry.exchange.grpc-web.injective.network:443")
+const injectiveClient = getClient()
 
-const autocompoundContract = 'inj1mjcg8a73904rj4w7t5qkgn0apua98n059nufma'
 async function tvl(_, _1, _2, { api }) {
-  const { total_supply } = await queryContract({ chain: api.chain, contract: "inj18luqttqyckgpddndh8hvaq25d5nfwjc78m56lc", data: { token_info: {} } })
-  api.add(ADDRESSES.injective.INJ, total_supply)
+  const { total_supply } = await queryContract({ chain: api.chain, contract: hinj, data: { token_info: {} } })
+  const { total_bonded } = await queryContract({ chain: api.chain, contract: autoCompound, data: { state: {} } })
```
Comment (g1nt0ki) [inline projects/hydro-protocol/index.js:19]: why are you including the INJ in autocompounder, arent they converted to hINJ, so already counted as part of hINJ supply/tvl?

File: projects/hydro-protocol/index.js [hunk 2]
```diff
@@ -1,10 +1,65 @@
...
+  const hdroCirculatingSupply = new BigNumber(await get("https://index.hydroprotocol.finance/circulating-supply"))
+  const marketInjQuantity = await loadMarketInjQuantity()
+  const hdroInjPrice = await loadHdroInjPrice()
+
+  api.add(
+    ADDRESSES.injective.INJ,
+    new BigNumber(total_supply)
+      .plus(new BigNumber(total_bonded))
+      .plus(new BigNumber(marketInjQuantity))
```
Comment (g1nt0ki) [inline projects/hydro-protocol/index.js:28]: why is this counted as your product tvl?

File: projects/hydro-protocol/index.js [hunk 3]
```diff
@@ -1,10 +1,65 @@
...
+      .plus(hdroCirculatingSupply.multipliedBy(hdroInjPrice))
```
Comment (g1nt0ki) [inline projects/hydro-protocol/index.js:29]: why is circulating supply of HDRO considered tvl? https://docs.llama.fi/list-your-project/what-to-include-as-tvl

File: projects/hydro-protocol/index.js (hunk 1 above)
Comment (hydro-developer) [reply projects/hydro-protocol/index.js:19]: No Auto-compound does not use hINJ. It uses INJ.
hINJ is used in LSD & Farm contract only

File: projects/hydro-protocol/index.js (hunk 2 above)
Comment (hydro-developer) [reply projects/hydro-protocol/index.js]: I fixed it to marketINJQuantity -> marketHdroQuantity

**Comment by llamatester** [comment]
Error while running adapter at projects/hydro-protocol: 
        
 
  
    
    
    Error: Protocol doesn't have total tvl, make sure to export a tvl key either on the main object or in one of the chains
    Truncated error stack:
    
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:192:11

File: projects/hydro-protocol/index.js (hunk 3 above)
Comment (hydro-developer) [reply projects/hydro-protocol/index.js]: Now it contains only locked hdro (market + gov stake)

File: projects/hydro-protocol/index.js (hunk 1 above)
Comment (g1nt0ki) [reply projects/hydro-protocol/index.js:19]: Ah, I am confused now, what does it do with the inj staked there/how does it compound?

File: projects/hydro-protocol/index.js (hunk 2 above)
Comment (g1nt0ki) [reply projects/hydro-protocol/index.js]: Locked hrdo can be counted under staking, unsure why you are trying to count hrdo/inj in order book as tvl

File: projects/hydro-protocol/index.js (hunk 1 above)
Comment (hydro-developer) [reply projects/hydro-protocol/index.js:19]: Auto-compound contract restake its inj every day.

Comment (g1nt0ki) [reply projects/hydro-protocol/index.js:19]: it restakes to hINJ ?

File: projects/hydro-protocol/index.js (hunk 2 above)
Comment (hydro-developer) [reply projects/hydro-protocol/index.js]: Because it is our liquidity

File: projects/hydro-protocol/index.js (hunk 1 above)
Comment (hydro-developer) [reply projects/hydro-protocol/index.js:19]: No

Comment (hydro-developer) [reply projects/hydro-protocol/index.js:19]: Auto-compound does not use hINJ

File: projects/hydro-protocol/index.js (hunk 2 above)
Comment (g1nt0ki) [reply projects/hydro-protocol/index.js]: Yes, but that is not what we consider as tvl, tvl is value locked into your protocol/contracts but the liquidity value belongs to the Dex/orderbook protocol

File: projects/hydro-protocol/index.js (hunk 1 above)
Comment (g1nt0ki) [reply projects/hydro-protocol/index.js:19]: It restakes using a different staking contract?

File: projects/hydro-protocol/index.js (hunk 2 above)
Comment (hydro-developer) [reply projects/hydro-protocol/index.js]: Ok thanks. I'm going to remove it

**Comment by llamatester** [comment]
Error while running adapter at : 
        
 
  > Please revert changes to package.json / package-lock.json

File: projects/hydro-protocol/index.js (hunk 1 above)
Comment (hydro-developer) [reply projects/hydro-protocol/index.js:30]: No auto-compound contract restake itself

Comment (hydro-developer) [reply projects/hydro-protocol/index.js:30]: Hydro-protocol provide two different type INJ staking.
First, LSD (hINJ) type that you know. If user stake INJ to LSD contract then it gives hINJ to user.
Second, Auto-compound (INJ) type. If user stake INJ to Auto-compound contract then it gives nothing to user. Auto-compound contract stake users' INJ and execute (claim inj reward) + (stake) messages every day.

Comment (g1nt0ki) [reply projects/hydro-protocol/index.js:30]: thanks for patiently explaining it

**Comment by llamatester** [comment]
Error while running adapter at : 
        
 
  > Please revert changes to package.json / package-lock.json

**Comment by llamatester** [comment]
Error while running adapter at projects/hydro-protocol: 
        
 
  
    
    
    Error: Protocol doesn't have total tvl, make sure to export a tvl key either on the main object or in one of the chains
    Truncated error stack:
    
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:192:11

**Comment by llamatester** [comment]
The adapter at projects/hydro-protocol exports TVL: 
        
 
 
    injective-staking         2.85 M
    injective                 73.28 M
    staking                   2.85 M
    
    total                    73.28 M

**Comment by g1nt0ki** [comment]
@hydro-developer thanks for the PR, I have refactored and merged it

**Pull request merged**